pkg/io/confio/read_defs.go

```diff
@@ -35,46 +40,146 @@ func ReadDefs(filename string) (*ir.ConfigDefs, error) {
+			nifUniqueName := scopingString(instanceUniqueName, *instance.NetworkInterfaces[i].Name)
+			nifDetails := ir.NifDetails{
+				NamedEntity: ir.NamedEntity(*instance.NetworkInterfaces[i].Name),
+				Instance:    ir.ID(scopingString(*instance.VPC.Name, *instance.Name)),
+				VPC:         ir.ID(*instance.VPC.Name),
+				IP:          ir.IPFromString(*instance.NetworkInterfaces[i].PrimaryIP.Address),
+			}
+			nifs[ir.ID(nifUniqueName)] = &nifDetails
+			instanceNifs[i] = ir.ID(nifUniqueName)
+		}
+		instanceDetails := ir.InstanceDetails{
+			NamedEntity: ir.NamedEntity(*instance.Name),
+			VPC:         ir.ID(*instance.VPC.Name),
+			Nifs:        instanceNifs,
+		}
+		instances[ir.ID(instanceUniqueName)] = &instanceDetails
+	}
+	return instances, nifs
+}
+
+func parseVPEs(config *configModel.ResourcesContainerModel) (vpes map[ir.ID]*ir.VPEDetails, vpeEndpoints map[ir.ID]*ir.VPEEndpointDetails) {
+	vpes = make(map[ir.ID]*ir.VPEDetails)
+	vpeEndpoints = make(map[ir.ID]*ir.VPEEndpointDetails)
+
+	for _, vpe := range config.EndpointGWList {
+		if *vpe.ResourceType == EndpointVPE {
+			uniqueVpeName := scopingString(*vpe.VPC.Name, *vpe.Name)
+			vpeDetails := ir.VPEDetails{
+				NamedEntity: ir.NamedEntity(*vpe.Name),
+				VPEEndpoint: []ir.ID{},
+				VPC:         ir.ID(*vpe.VPC.Name),
+			}
+			vpes[ir.ID(uniqueVpeName)] = &vpeDetails
 		}
-		instanceToNIF[*instance.Name] = nifs
 	}
 
-	vpeToIP := make(map[string]ir.IP)
 	for _, subnet := range config.SubnetList {
 		for _, r := range subnet.ReservedIps {
 			if t, ok := r.Target.(*vpcv1.ReservedIPTarget); ok && t != nil && r.Address != nil {
-				if r.ResourceType != nil && *t.ResourceType == "endpoint_gateway" && t.Name != nil {
-					vpeToIP[*t.Name] = ir.IPFromString(*r.Address)
+				if r.ResourceType != nil && *t.ResourceType == EndpointVPE && t.Name != nil {
+					VPEName := ir.ID(scopingString(*subnet.VPC.Name, *t.Name))
+					subnetName := ir.ID(scopingString(*subnet.VPC.Name, *subnet.Name))
+					uniqueVpeEndpointName := scopingString(string(VPEName), *r.Name)
+					vpeEndpointDetails := ir.VPEEndpointDetails{
+						NamedEntity: ir.NamedEntity(*r.Name),
+						VPEName:     VPEName,
+						Subnet:      subnetName,
+						IP:          ir.IPFromString(*r.Address),
+						VPC:         vpes[VPEName].VPC,
+					}
+					vpeEndpoints[ir.ID(uniqueVpeEndpointName)] = &vpeEndpointDetails
+					vpe := vpes[VPEName]
+					vpe.VPEEndpoint = append(vpe.VPEEndpoint, ir.ID(uniqueVpeEndpointName))
+					vpes[VPEName] = vpe
 				}
 			}
 		}
 	}
 
-	addressPrefixes := make([]ir.CIDR, 0)
-	for _, vpc := range config.VpcList {
-		for _, addressPrefix := range vpc.AddressPrefixes {
-			addressPrefixes = append(addressPrefixes, ir.CidrFromString(*addressPrefix.CIDR))
+	return vpes, vpeEndpoints
+}
+
+func validateVpcs(vpcs map[ir.ID]*ir.VPCDetails) error {
+	for vpcName1, vpcDetails1 := range vpcs {
+		for vpcName2, vpcDetails2 := range vpcs {
+			if vpcName1 == vpcName2 {
```

Comment:
```suggestion
			if vpcName1 >= vpcName2 {
```
So you don't check a pair of VPCs twice (VPC-A vs. VPC-B and VPC-B vs. VPC-A)